src/Microsoft.Health.SqlServer/IAccessTokenHandler.cs

```diff
@@ -1,20 +1,30 @@
-﻿// -------------------------------------------------------------------------------------------------
+// -------------------------------------------------------------------------------------------------
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
+using Microsoft.Health.SqlServer.Configs;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.Health.SqlServer;
 
 public interface IAccessTokenHandler
 {
+    /// <summary>
+    /// Determines the type of SqlServiceAuthenticationType. This correlates with the type of token credential that will be used by the implementation.
+    /// </summary>
+    SqlServerAuthenticationType AuthenticationType { get; }
```

Comment:
What is the point of this `interface`? I don't see any code use this abstraction. It seems we would be better served simply using the concrete derived types as the services in the DI container, instead of using this `interface`. E.g. remove `IAccessTokenHandler` and simply register both `ManagedIdentityTokenAccessHandler` and `WorkloadIdentityAccessHandler` separately.

Reply:
I am using this in SqlServerBaseRegistrationExtensions. So depends on the type of AuthenticationType the workflow will determine which Tokencredential type to use.